### PR TITLE
Replace current toolbar with list-builder's multiselect toolbar

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Addresses #13

### CODE REVIEW

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.spec.ts [hunk 1]
```diff
@@ -594,19 +603,22 @@ describe('List View Checklist Component', () => {
       });
     }));
 
-    it('should select all and clear all properly', fakeAsync(() => {
+    fit('should select all and clear all properly when not all items are displayed', fakeAsync(() => {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/list-view-checklist/list-view-checklist.component.spec.ts]: `fit`

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.ts [hunk 2]
```diff
@@ -368,4 +299,13 @@ export class SkyListViewChecklistComponent extends ListViewComponent implements
       }
     this.disableToolbar(isSelected);
   }
+
+  private showSelectedValuesEqual(prev: ListFilterModel[], next: ListFilterModel[]) {
+    const prevShowSelectedFilter = prev.find(filter => filter.name === 'show-selected');
+    const nextShowSelectedFilter = next.find(filter => filter.name === 'show-selected');
+    if (prevShowSelectedFilter && nextShowSelectedFilter) {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/list-view-checklist/list-view-checklist.component.ts:338]: I can't find it in our style guide; however, I believe @Blackbaud-SteveBrush mentioned once that we should have line breaks on both sides of a block of code like the `if block` here.

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.ts [hunk 3]
```diff
@@ -252,82 +259,6 @@ export class SkyListViewChecklistComponent extends ListViewComponent implements
     this.dispatcher.next(new ListSelectedSetItemsSelectedAction([item.id], true, true));
   }
 
-  public clearSelections() {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/list-view-checklist/list-view-checklist.component.ts:255]: These are public functions and thus would be breaking changes...... I think we have to leave them even if we mark them as deprecated.

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.ts [hunk 4]
```diff
@@ -161,18 +161,29 @@ export class SkyListViewChecklistComponent extends ListViewComponent implements
...
+  }
+
   public ngOnChanges(changes: SimpleChanges) {
     if (changes['showOnlySelected'] &&
       changes['showOnlySelected'].currentValue !== changes['showOnlySelected'].previousValue) {
       this.reapplyFilter(changes['showOnlySelected'].currentValue);
     }
   }
 
-  public changeVisibleItems(change: SkyCheckboxChange) {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/list-view-checklist/list-view-checklist.component.ts]: See my other comment about public functions

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.ts [hunk 5]
```diff
@@ -252,82 +259,6 @@ export class SkyListViewChecklistComponent extends ListViewComponent implements
...
-        this.dispatcher
-          .next(new ListSelectedSetItemsSelectedAction(items.map(item => item.id), false, false));
-
-        if (this.showOnlySelected) {
-          this.reapplyFilter(this.showOnlySelected);
-        }
-      });
-  }
-
-  public selectAll() {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/list-view-checklist/list-view-checklist.component.ts:268]: See my other comment about public functions

**Review by Blackbaud-TrevorBurch — CHANGES_REQUESTED** [review]
A few things to think about :)

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.spec.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-checklist/list-view-checklist.component.spec.ts]: good catch. removed.

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.ts (hunk 3 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-checklist/list-view-checklist.component.ts:255]: I guess you're right. I've put them back with deprecated decorators.

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.ts (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-checklist/list-view-checklist.component.ts:338]: done.

File: src/app/public/modules/list-view-checklist/list-view-checklist.component.ts (hunk 3 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-checklist/list-view-checklist.component.ts:255]: I've also confirmed these methods still work properly, so if a consumer is still using them, they'll be OK.

**Review by Blackbaud-TrevorBurch — APPROVED** [review]
LGTM

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**